CDCgov/prime-reportstream
# 16140: manually populate poison queue when a step encounters an error

This PR updates the FHIR function error handling to manually add most error cases to the poison queue

Test Steps:
This is mostly covered by automation, but to test locally.

1. In FHIRConverter, update the `doWork` to throw a Runtime exception
2. Verify that the poison queue is populated and the message is not retried

## Changes
- Updates how the FHIR functions run each FHIR engine so that only data access exceptions are retried and all other are manually moved to the poison queue

## Checklist

### Testing

### Process

## Linked Issues
- #16140 

## To Be Done


## Specific Security-related subjects a reviewer should pay specific attention to
